woocommerce/woocommerce-legacy-rest-api
# Add sanitization for input data

Sanitization is added for all the input data received via `$_GET` and `$_SERVER` arrays, with the exception of `$_GET['_jsonp']` because these are already verified with `wp_check_jsonp_callback`.

## Testing instructions

Test that the extension continues working as per the instructions in https://github.com/woocommerce/woocommerce-legacy-rest-api/pull/1, but focusing in the following in particular:

1. Test a request that contains characters, numbers and at least one slash in the path, e.g. `orders/<order id>`.
1. Test with the regular OAuth authentication.
2. Test with basic authentication, passing the consumer key and secret as the user and password, respectively, as a HTTP header in the request: `Authorization: BASIC <base64 of key:secret>`.
3. Test with basic authentication, passing the consumer key and secret as query string arguments (`consumer_key` and `consumer_secret`, respectively).
4. Test supplying the request method in a `_method` key in the query string (e.g. issue a `POST` request to a GET endpoint with `?_method=GET` in the URL).
5. Test supplying the request method within a `X-HTTP-Method-Override` header in the request (e.g. issue a `POST` request to a GET endpoint with `X-HTTP-Method-Override: GET` added to the request headers).
6. Test with `v1`,  `v2` and `v3` endpoints.
7. Check if there's still any `$_GET` or `$_SERVER` variable that is still not sanitized. Keep in mind that the entire `$_GET` array is copied to a `$params` property in `class-wc-api-server.php`.

In order to test basic authentication you need to either use a SSL-enabled server or (probably easier) to temporarily tweak the code as follows: in the `authenticate` function in the `class-wc-api-authentication.php` files (there's one per API version) replace the line `if( is_ssl() ) {` with `if(true) {`.
